cachi2/core/package_managers/yarn/main.py

```diff
@@ -73,21 +75,25 @@ def _configure_yarn_version(project: Project) -> None:
             ),
         )
 
-    if (
-        yarn_path_version
-        and package_manager_version
-        and yarn_path_version != package_manager_version
-    ):
-        raise PackageRejected(
-            (
-                f"Mismatch between the yarn versions specified by yarnPath (yarn@{yarn_path_version}) "
-                f"and packageManager (yarn@{package_manager_version})"
-            ),
-            solution=(
-                "Ensure that the versions of yarn specified by yarnPath in .yarnrc.yml and "
-                "packageManager in package.json agree"
-            ),
-        )
+    if yarn_path_version and package_manager_version:
+        if yarn_path_version != package_manager_version:
+            raise PackageRejected(
+                (
+                    f"Mismatch between the yarn versions specified by yarnPath (yarn@{yarn_path_version}) "
+                    f"and packageManager (yarn@{package_manager_version})"
+                ),
+                solution=(
+                    "Ensure that the versions of yarn specified by yarnPath in .yarnrc.yml and "
+                    "packageManager in package.json agree"
+                ),
+            )
+    else:
```

Comment:
This else here introduces a flaw: the version will not be verified if both `yarn_path_version` and `package_manager_version` are set, and it should.

You can test by getting a sample yarn repo and do `yarn set version 2.0.0`, then try to process it with Cachi2.

Reply:
Doh! Very true.